[CI] TransportVersionTests testVersionComparison failing

**Build Scans:**
- [elasticsearch-periodic #11545 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/hcdbozy33dz3s)
- [elasticsearch-periodic-platform-support #11513 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/653wfcb4clvha)
- [elasticsearch-pull-request #108064 / part-1](https://gradle-enterprise.elastic.co/s/34ztc2n7xctuw)
- [elasticsearch-pull-request #108049 / part-1](https://gradle-enterprise.elastic.co/s/4rzm4fgtr2jqw)
- [elasticsearch-pull-request #107945 / part-1](https://gradle-enterprise.elastic.co/s/gfxpl6gvin25s)
- [elasticsearch-periodic-platform-support #11506 / rhel-10_platform-support-unix](https://gradle-enterprise.elastic.co/s/nhjyme3gugdxa)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.TransportVersionTests.testVersionComparison" -Dtests.seed=CA811B984C26C18E -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=zu-ZA -Dtests.timezone=Asia/Aqtau -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.TransportVersionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testVersionComparison'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 6 failures in test testVersionComparison (2.3% fail rate in 264 executions)
- [main] 3 failures in step part-1 (1.9% fail rate in 159 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.9% fail rate in 157 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -488,9 +488,6 @@ tests:
 - class: org.elasticsearch.simdvec.Int7SQVectorScorerFactoryTests
   method: testBulk
   issue: https://github.com/elastic/elasticsearch/issues/139079
-- class: org.elasticsearch.TransportVersionTests
-  method: testVersionComparison
-  issue: https://github.com/elastic/elasticsearch/issues/139122
 - class: org.elasticsearch.upgrades.IndexSortUpgradeIT
   method: testIndexSortForNumericTypes {upgradedNodes=1}
   issue: https://github.com/elastic/elasticsearch/issues/139127

--- a/server/src/test/java/org/elasticsearch/TransportVersionTests.java
+++ b/server/src/test/java/org/elasticsearch/TransportVersionTests.java
@@ -39,7 +39,11 @@ public class TransportVersionTests extends ESTestCase {
             TransportVersion.minimumCompatible(),
             TransportVersionUtils.getPreviousVersion(TransportVersion.current())
         );
-        TransportVersion newer = TransportVersionUtils.randomVersionBetween(random(), older, TransportVersion.current());
+        TransportVersion newer = TransportVersionUtils.randomVersionBetween(
+            random(),
+            TransportVersionUtils.getNextVersion(older),
+            TransportVersion.current()
+        );
         assertThat(older.before(newer), is(true));
         assertThat(older.before(older), is(false));
         assertThat(newer.before(older), is(false));

